api/integrations/google/oauth.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import argparse
+import datetime
+import httplib2
+import os
+
+# Uses Google's Client Library
+#   pip install --upgrade google-api-python-client
+from oauth2client import client, tools
+from oauth2client.file import Storage
+
+# Before running this, make sure to register a new
+# project in the Google Developers Console
+# (https://console.developers.google.com/start)
+# and download the "client secret" JSON file.
+#
+# You'll have to specify its location with --secret-path.
+
+# This parent argparser is used because it already contains
+# the arguments that Google's Client Library method "tools.run_flow"
+# supports.
```

Comment:
No information on specific scopes is given here, because this script is intented for usage with any other integration that uses Google services. Thus, each bot/integration will need different scopes, depending on its needs. That also explains why it's in `api/integrations/google/`, and not with my bot's code (this was initially created by @fazerlicourice7).

If you take a look at the bot's code itself (`contrib_bots/gcalendar.py`), you'll see that the correct scope is indicated in the setup comments:

```
# Modify this to match where your credentials are stored.
# It should be the same path you used in the
# '--credential_path' argument, when running the script
# in "api/integrations/google/oauth.py".
#
# $ python api/integrations/google/oauth.py \
#   --secret_path <your_secret_file_path> \
#   --credential_path <your_credential_path> \
#   -s https://www.googleapis.com/auth/calendar
#
# If you didn't specify any, it should be in the default
# path (~/.google_credentials.json)
CREDENTIAL_PATH = '~/.google_credentials.json'
```

As well as later on, in the bot's `usage()` docstring:

```python
return '''
            This plugin will allow users to create events
            in their Google Calendar. Users should preface
            messages with "@gcalendar".

            Before running this, make sure to register a new
            project in the Google Developers Console
            (https://console.developers.google.com/start/api?id=calendar),
            download the "client secret" as a JSON file, and run
            "api/integrations/google/oauth.py" for the Calendar
            read-write scope
            (https://www.googleapis.com/auth/calendar).
            '''
```